### PR TITLE
DYN-3070-CodeCoverage-PythonNodeModelsWpf

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose
I added several test methods for the next events in the ScriptEditorWindow.xaml.cs file:
- OnTextAreaTextEntering
- OnSaveClicked
- OnRevertClicked
- OnRunClicked
- OnMoreInfoEvent

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 

### FYIs

@alfredo-pozo

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
I ran the DynamoSelfServe job and was successful
image

File: test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs [hunk 1]
```diff
@@ -96,6 +104,169 @@ public void CanChangeEngineFromScriptEditorDropDown()
...
+            string pythonCode = "from System.Collections import ArrayList";
+            Open(@"core\python\python.dyn");
+
+            var nodeView = NodeViewWithGuid("3bcad14e-d086-4278-9e08-ed2759ef92f3");
+            var nodeModel = nodeView.ViewModel.NodeModel as PythonNodeBase;
+            Assert.NotNull(nodeModel);
+
+            var scriptWindow = EditPythonCode(nodeView, View);
+            var codeEditor = FindCodeEditor(scriptWindow);
+            scriptWindow.editText.TextArea.TextEntering += TextArea_TextEntering;
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs:123]: unsubscribe?

Comment (RobertGlobant20) [reply test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs:123]: @QilongTang I added the code for unsubscribing the event in the next commit:

https://github.com/DynamoDS/Dynamo/pull/11137/commits/362fc84368f0c7b5537c8378491a724f7a002a67

File: test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs [hunk 2]
```diff
@@ -96,6 +104,170 @@ public void CanChangeEngineFromScriptEditorDropDown()
...
+            var nodeView = NodeViewWithGuid("3bcad14e-d086-4278-9e08-ed2759ef92f3");
+            var nodeModel = nodeView.ViewModel.NodeModel as PythonNodeBase;
+            Assert.NotNull(nodeModel);
+
+            //Get the Watch node so we can check the content later
+            var watchOUT = Model.CurrentWorkspace.NodeFromWorkspace<Watch>("714838b43a114f5f93cfe2f1d26092cf");
+
+            var scriptWindow = EditPythonCode(nodeView, View);
+            var codeEditor = FindCodeEditor(scriptWindow);
+            codeEditor.Text = codeEditor.Text.Replace("Autodesk", "Softdesk").Replace("1982", "1997");
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs:247]: hmm what does this line do?

File: test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs [hunk 3]
```diff
@@ -96,6 +104,170 @@ public void CanChangeEngineFromScriptEditorDropDown()
...
+            DispatcherUtil.DoEvents();
+
+            //Check that now we are showing a extension tab (DocumentationBrowser)
+            Assert.That(this.View.ExtensionTabItems.Count, Is.EqualTo(1));
+
+            var docBrowser = this.View.ExtensionTabItems
+                            .Where(x => x.Content.GetType().Equals(typeof(DocumentationBrowserView)))
+                            .FirstOrDefault();
+
+            //Validate hat the DocmentationBrowser extension tab is valid
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs:266]: validate that?

File: test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs (hunk 2 above)
Comment (RobertGlobant20) [reply test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs:249]: @QilongTang 
This line is replacing the python dictionary values so later after re-running we will see the changes updated in the Watch node. I added an extra comment clarifying this info.

File: test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs (hunk 3 above)
Comment (RobertGlobant20) [reply test/DynamoCoreWpfTests/PythonNodeCustomizationTests.cs]: @QilongTang 
Fixed in the comment: https://github.com/DynamoDS/Dynamo/pull/11137/commits/d86ba2c296c92207d573903831a776355bf1ba67

**Pull request merged**